drsm/api.go

```diff
@@ -68,21 +68,23 @@ func (d *Drsm) AllocateInt32ID() (int32, error) {
 	mutex.Lock()
 	defer mutex.Unlock()
 	if d.mode == ResourceDemux {
-		logger.DrsmLog.Debugln("demux mode can not allocate Resource index")
+		logger.DrsmLog.Errorln("demux mode can not allocate Resource index")
 		err := fmt.Errorf("demux mode does not allow Resource Id allocation")
 		return 0, err
 	}
 	for _, c := range d.localChunkTbl {
 		if len(c.FreeIds) > 0 {
-			return c.AllocateIntID(), nil
+			return c.AllocateIntID()
 		}
 	}
+	// None of the Chunk has freeIds. Allocate new Chunk
 	c, err := d.GetNewChunk()
 	if err != nil {
-		err := fmt.Errorf("ids not available")
+		logger.DrsmLog.Errorln("failed to allocate new Chunk")
+		err := fmt.Errorf("failed to allocate new Chunk")
```

Comment:
The code will never enter this `if` statement because `d.GetNewChunk()` will always return `err=nil`, correct?
From `chunk.go`, there is no other return besides `return c, nil`
```go
func (d *Drsm) GetNewChunk() (*chunk, error) {
...
...
	// why we are not adding in global table right away???
	return c, nil
}
```

Reply:
Agree..Just because getNewChunk is not giving up at some poinnt in time.. if it has any timeout or retry limit then the function will return error.